src/Symfony/Component/Serializer/Serializer.php

```diff
@@ -138,16 +136,15 @@ public function normalize($data, $format = null, array $context = array())
 
             return $normalized;
         }
+
         if (is_object($data)) {
-            return $this->normalizeObject($data, $format, $context);
-        }
-        if (is_array($data)) {
-            foreach ($data as $key => $val) {
-                $data[$key] = $this->normalize($val, $format, $context);
+            if (!$this->normalizers) {
```

Comment:
To trigger an exception if `$data` is an object and `$this->normalizer` is not empty.